gst/nnstreamer/elements/gsttensor_generator.c

```diff
@@ -0,0 +1,415 @@
+}
+
+/**
+ * @brief Function to finalize instance.
+ */
+static void
+gst_tensor_generator_finalize (GObject * object)
+{
+  G_OBJECT_CLASS (parent_class)->finalize (object);
+}
+
+/**
+ * @brief Setter for tensor_generator properties.
+ */
+static void
+gst_tensor_generator_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  UNUSED (value);
+  switch (prop_id) {
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Getter for tensor_generator properties.
+ */
+static void
+gst_tensor_generator_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec)
+{
+  UNUSED (value);
+  switch (prop_id) {
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Handle state transition.
+ */
+static GstStateChangeReturn
+gst_tensor_generator_change_state (GstElement * element, GstStateChange transition)
+{
+  GstTensorGenerator *self;
+  GstStateChangeReturn ret;
+
+  self = GST_TENSOR_GENERATOR (element);
+
+  switch (transition) {
+    case GST_STATE_CHANGE_READY_TO_PAUSED:
+      gst_collect_pads_start (self->collect);
+      break;
+    case GST_STATE_CHANGE_PAUSED_TO_READY:
+      gst_collect_pads_stop (self->collect);
+      break;
+    default:
+      break;
+  }
+
+  ret = GST_ELEMENT_CLASS (parent_class)->change_state (element, transition);
+
+  return ret;
+}
+
+static gboolean
+gst_generator_signal_query (GstPad * pad, GstObject * parent, GstQuery * query)
+{
+  gboolean res = FALSE;
+  GstCaps *caps;
+
+  switch (GST_QUERY_TYPE (query)) {
+    case GST_QUERY_CAPS:
+      caps = gst_caps_new_simple ("signals", "framerate", GST_TYPE_FRACTION, 0, 1, NULL);
+      gst_query_set_caps_result (query, caps);
+      gst_caps_unref (caps);
+      return TRUE;
+    case GST_QUERY_ACCEPT_CAPS:
+      gst_query_set_accept_caps_result (query, TRUE);
+      return TRUE;
+    default:
+      res = gst_pad_query_default (pad, parent, query);
+      break;
+  }
+
+  return res;
+}
+
+/**
+ * @brief Handle event on src pad.
+ */
+static gboolean
+gst_tensor_generator_src_event (GstPad * pad, GstObject * parent, GstEvent * event)
+{
+  g_return_val_if_fail (event != NULL, FALSE);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_SEEK:
+      /* disable seeking */
+      gst_event_unref (event);
+      return FALSE;
+    default:
+      break;
+  }
+
+  return gst_pad_event_default (pad, parent, event);
+}
+
+static gpointer
+tensor_generator_signal_activate (GstTensorGenerator *self)
+{
+  GstBuffer *dummy;
+  GstMemory *mem;
+  guint8 *data;
+
+  dummy = gst_buffer_new ();
+  data = (guint8 *) g_malloc0 (1);
+  mem = gst_memory_new_wrapped (0, data, 1, 0 ,1, data, g_free);
+  gst_buffer_append_memory (dummy, mem);
+
+  gst_pad_push (self->srcpad_signal, dummy);
+  gst_pad_push_event (self->srcpad_signal, gst_event_new_eos ());
```

Comment:
`gst_pad_push` is ended when `gst_buffer_unref (gst_collect_pads_pop (self->collect, data_signal));` is called (LINE 412 now).
So eos event is sent right after  `gst_pad_push`.
Please let me know if I'm using g_thread incorrectly